storage/src/mmr/mem.rs

```diff
@@ -358,6 +379,33 @@ impl<H: CHasher> Mmr<H> {
         panic!("invalid pos {}:{}", pos, self.size());
     }
 
+    /// Batch update the digests of multiple retained leaves using multiple threads.
+    fn update_leaf_parallel<T: AsRef<[u8]> + Sync>(
+        &mut self,
+        hasher: &mut impl Hasher<H>,
+        pool: &RThreadPool,
```

Comment:
I think we should only be using our `runtime` pool here?

Reply:
Pushed the unwrapping / cloning of the ThreadPool within these functions so they don't need to be passed in anymore.  (Also tried implementing pool.install on the runtime ThreadPool but then we end up with an immutable reference to self, preventing the mutable use of self within the rayon closure).